main.cpp

```diff
@@ -7,39 +8,63 @@
 // main file
 int main(int argc, char **argv) {
     try {
-        // Define allowable filters
-        const char *filtersOptions {"begrs"};
-
-        // Get filter flag
-        char filter {static_cast<char>(getopt(argc, argv, filtersOptions))};
-
-        // Check validity
-        if (filter == '?') {
-            throw std::runtime_error {"Invalid filter."};
+        // Ensure proper usage
+        if (argc != 3) {
+            throw std::runtime_error {"Usage: filter [flag] infile outfile"};
         }
 
-        // Ensure only one filter
-        if (getopt(argc, argv, filtersOptions) != -1) {
-            throw std::runtime_error {"Only one filter allowed."};
-        }
+        // Define allowable filters
+        std::string filtersOptions {"bBeEgGrRsS"};
+        // To read input option(s) from terminal
+        std::string instr;
+        
+        // Get filter option
+        char filter {'\0'};
+
+        while(true) {
+            // Filter menu
+            std::cout << "\n\n FILTER    : OPTION" << std::endl;
+            std::cout << " ------------------" << std::endl;
+            std::cout << " Blur        : b\n Edge        : e\n GrayScale   : g\n Reflect     : r\n Sepia       : s\n" << std::endl;
+            std::cout << " To QUIT press    : [q]" << std::endl;
+            std::cout << "Enter option :";
+            std::cin  >> instr;
+
+            // Ensure only one filter
+            if (instr.length() != 1) {
+                std::cout << "Only one option allowed.\nTry again..." << std::endl;
+                continue;
+            }
 
-        // Ensure proper usage
-        if (argc != optind + 2) {
-            throw std::runtime_error {"Usage: filter [flag] infile outfile"};
+            filter = instr.at(0);
+            
+            // check validates option
+            if (filtersOptions.find(filter) != std::string::npos) {
+                std::cout << "Good choice" << std::endl;
```

Comment:
Can we have more descriptive information instead of `Good Choice` ? @chandrakant100 @Ank1taS Otherwise everything is fine from my side.

Reply:
How about `Applied filter on [File name] successfully`? But in that case, the code will be more extended if we want to specify which filter is applied. Alternatively, we can specify the same statement in the switch block in each case as ` Applied [filter name] filter on [file name] successfully`.

Reply:
`Applied [filter name] filter on [file name] successfully` will be the correct one to describe what happened to the file.

Reply:
ok I will update pr

Reply:
> `Applied [filter name] filter on [file name] successfully` will be the correct one to describe what happened to the file.

This sounds more friendly